client/state/reducer.js

```diff
@@ -92,6 +94,9 @@ const reducers = {
 	userProfileLinks,
 	userSettings,
 	users,
+	// TODO: remove this once admin-menu state is consumed by UI components
+	// to allow it to be loaded as modularized state.
+	...( config.isEnabled( 'nav-unification' ) && { adminMenu } ),
```

Comment:
I suggest doing things differently here. All you're looking to do is to ensure that the admin menu state is loaded, even if nothing is making use of it, correct?

If so, I'd remove all existing changes to this file, and instead just do the following a bit further down:

```js
if ( config.isEnabled( 'nav-unification' ) ) {
  import( 'state/admin-menu/init' ); 
}
```

That explicitly initialises admin menu state as if it were being loaded by a selector or action creator. It also keeps that code outside of the critical path, because it's a dynamic import, rather than a static one which always gets bundled regardless of the outcome of the condition.

The only downside is that a dynamic import is asynchronous. However, since there's no actual code depending on it that could run before the import finishes, that's not a problem.

Reply:
Thanks for this. Makes totaly sense. Have updated.